include/alpaka/core/RemoveRestrict.hpp

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2021 Rene Widera
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/BoostPredef.hpp>
+
+namespace alpaka
+{
+    //! Removes __restrict__ from a type
+    template<typename T>
+    struct remove_restrict
+    {
+        using type = T;
+    };
+
+#if BOOST_COMP_MSVC
+    template<typename T>
+    struct remove_restrict<T __restrict>
+    {
+        using type = T;
+    };
+#else
```

Comment:
For MSVC, you need to make the template only match pointers.

```suggestion
#if BOOST_COMP_MSVC
    template<typename T>
    struct remove_restrict<T* __restrict>
    {
        using type = T;
    };
#else
```